plugins/auth-backend-module-guest-provider/src/createGuestAuthFactory.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SignInResolverFactory } from '@backstage/plugin-auth-node';
+import type {
+  AuthProviderFactory,
+  ProfileTransform,
+  SignInResolver,
+} from '@backstage/plugin-auth-node';
+import { createGuestAuthRouteHandlers } from './createGuestAuthRouteHandlers';
+import { guestResolver } from './resolvers';
+
+const defaultTransform: ProfileTransform<{}> = async () => {
+  return {
+    profile: {
+      displayName: 'Guest',
+    },
+  };
+};
+
+/** @public */
+export function createGuestAuthProviderFactory(options?: {
```

Comment:
I think the existing `createProxyAuthProviderFactory` should do it here tbh, and you can avoid adding all of these.

Reply:
😅 that makes this implementation a lot simpler